Metricbeat filters cause panic when removing events

- Version: master
- Operating System: darwin

Starting metricbeat master with this config:

```
metricbeat.modules:
- module: system
  enabled: true
  period: 1m
  metricsets:
    - filesystem
  filters:
    - drop_event.when.regexp.mount_point: '^/(sys|cgroup|proc|dev|etc|host|lib)($|/)'

output.elasticsearch.hosts: ["localhost:9200"]
```
Causes a panic in the master branch:

```
2017/05/16 12:35:21.928875 beat.go:262: INFO metricbeat start running.
panic: assignment to entry in nil map

goroutine 53 [running]:
github.com/elastic/beats/libbeat/publisher.(*client).annotateEvent(0xc42031b140, 0x0)
	/Users/tsg/src/github.com/elastic/beats/libbeat/publisher/client.go:180 +0xe0
github.com/elastic/beats/libbeat/publisher.(*client).PublishEvent(0xc42031b140, 0x0, 0x0, 0x0, 0x0, 0x1)
	/Users/tsg/src/github.com/elastic/beats/libbeat/publisher/client.go:99 +0x50
github.com/elastic/beats/metricbeat/mb/module.PublishChannels.func1(0xc420318c00)
	/Users/tsg/src/github.com/elastic/beats/metricbeat/mb/module/publish.go:25 +0xd4
created by github.com/elastic/beats/metricbeat/mb/module.PublishChannels
	/Users/tsg/src/github.com/elastic/beats/metricbeat/mb/module/publish.go:32 +0xec
```

But works with 5.4.1

The issue seems related to the filter, which results in a `nil` event.

--- a/CHANGELOG.asciidoc
+++ b/CHANGELOG.asciidoc
@@ -40,6 +40,7 @@ https://github.com/elastic/beats/compare/v6.0.0-alpha1...master[Check the HEAD d
 *Metricbeat*
 - Fix a debug statement that said a module wrapper had stopped when it hadn't. {pull}4264[4264]
 - Use MemAvailable value from /proc/meminfo on Linux 3.14. {pull}4316[4316]
+- Fix panic when events were dropped by filters. {issue}4327[4327]
 
 *Packetbeat*
 

--- a/metricbeat/mb/module/wrapper.go
+++ b/metricbeat/mb/module/wrapper.go
@@ -340,10 +340,13 @@ func (r *eventReporter) ErrorWith(err error, meta common.MapStr) bool {
 		return false
 	}
 
-	if !writeEvent(r.done, r.out, event) {
-		return false
+	if event != nil { // event can be nil if it was dropped by processors
+		if !writeEvent(r.done, r.out, event) {
+			return false
+		}
+		r.msw.stats.events.Add(1)
 	}
-	r.msw.stats.events.Add(1)
+
 	return true
 }
 

